UnitTests/OIDTVAuthorizationRequestTests.m

```diff
@@ -0,0 +1,198 @@
+  NSString *bodyString = [[NSString alloc] initWithData:urlRequest.HTTPBody
+                                               encoding:NSUTF8StringEncoding];
+  NSArray *bodyParameterStrings = [bodyString componentsSeparatedByString:@"&"];
+  
+
+  NSMutableDictionary<NSString *, NSString *> *bodyParameters = [[NSMutableDictionary alloc] init];
+
+  for (NSString *paramString in bodyParameterStrings) {
+    NSArray *components = [paramString componentsSeparatedByString:@"="];
+    NSLog(@"%@:%@", components[0], components[1]);
+    
+    [bodyParameters setObject:components[1] forKey:components[0]];
+  }
+
+  return bodyParameters;
+}
+
+- (void)testInitializer {
+  OIDTVAuthorizationRequest *authRequest =
+      [[OIDTVAuthorizationRequest alloc] initWithConfiguration:[self testServiceConfiguration]
+                                                      clientId:kTestClientID
+                                                  clientSecret:kTestClientSecret
+                                                        scopes:nil
+                                          additionalParameters:nil];
+
+  XCTAssertEqualObjects(authRequest.responseType, OIDResponseTypeCode);
+  XCTAssertEqualObjects(authRequest.redirectURL, [[NSURL alloc] init]);
+}
+
+- (void)testURLRequestBasicClientAuth {
+  OIDTVAuthorizationRequest *authRequest =
+      [[OIDTVAuthorizationRequest alloc] initWithConfiguration:[self testServiceConfiguration]
+                                                      clientId:kTestClientID
+                                                  clientSecret:kTestClientSecret
+                                                        scopes:nil
+                                          additionalParameters:nil];
+
+  NSURLRequest *urlRequest = [authRequest URLRequest];
+
+  XCTAssertEqualObjects([urlRequest HTTPMethod], kHTTPPost);
+  XCTAssertEqualObjects([urlRequest valueForHTTPHeaderField:kHTTPContentTypeHeaderKey],
+                        kHTTPContentTypeHeaderValue);
+  XCTAssertEqualObjects(urlRequest.URL.absoluteString, kInitializerTestTVAuthEndpoint);
+
+  NSDictionary<NSString *, NSString *> *bodyParameters =
+      [self bodyParametersFromURLRequest:urlRequest];
+  NSDictionary<NSString *, NSString *> *expectedParameters = @{@"client_id" : kTestClientID};
+
+  XCTAssertEqualObjects(bodyParameters, expectedParameters);
+}
+
+- (void)testURLRequestScopes {
+  OIDTVAuthorizationRequest *authRequest =
+      [[OIDTVAuthorizationRequest alloc] initWithConfiguration:[self testServiceConfiguration]
+                                                      clientId:kTestClientID
+                                                  clientSecret:kTestClientSecret
+                                                        scopes:@[ kTestScope, kTestScopeA ]
+                                          additionalParameters:nil];
+
+  NSURLRequest *urlRequest = [authRequest URLRequest];
```

Comment:
Same.

Reply:
in this case URLRequest is an instance method that does some logic in TVAuthorizationRequest